### PR TITLE
add refcoco to llava

### PR DESCRIPTION
**Pull request by LKJacky**
Add refcoco datasets for llava finetune.

It contains three parts.
1. datasets 
2. script to convert refcoco format to json format
3. script to eval refcoco grounding.

### CODE REVIEW

File: docs/en/user_guides/dataset_prepare.md
Comment (LZHgrla) [inline docs/en/user_guides/dataset_prepare.md:1]: Please add the data convert step (how to use tools/convert_refcoco) here.
Also, please update the zh_cn docs.

File: xtuner/configs/llava/vicuna_7b_v15_clip_vit_large_p14_336/finetune/llava_vicuna_7b_v15_qlora_clip_vit_large_p14_336_lora_e1_gpu8_finetune_refcoco.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+visual_encoder_name_or_path = 'openai/clip-vit-large-patch14-336'
+# Specify the pretrained pth
+pretrained_pth = (
+    'work_dirs/models--xtuner--llava-v1.5-7b-xtuner-pretrain/epoch_1.pth'  # noqa: E501
+)
+
+# Data
+data_root = './data/llava_data/'
+data_path = data_root + 'LLaVA-Instruct-150K/llava_v1_5_mix665k.json'
+refcoco_path = data_root + 'data/llava_data/RefCOCOJson/train.json'
```
Comment (LZHgrla) [inline xtuner/configs/llava/vicuna_7b_v15_clip_vit_large_p14_336/finetune/llava_vicuna_7b_v15_qlora_clip_vit_large_p14_336_lora_e1_gpu8_finetune_refcoco.py]: Please double-check the refcoco_path.

File: xtuner/configs/llava/vicuna_7b_v15_clip_vit_large_p14_336/finetune/llava_vicuna_7b_v15_qlora_clip_vit_large_p14_336_lora_e1_gpu8_finetune_refcoco.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+    #     type=EvaluateChatHook,
+    #     tokenizer=tokenizer,
+    #     image_processor=image_processor,
+    #     every_n_iters=evaluation_freq,
+    #     evaluation_inputs=evaluation_inputs,
+    #     evaluation_images=evaluation_images,
+    #     system=SYSTEM,
+    #     prompt_template=prompt_template,
+    #     max_new_tokens=20,
+    # ),
```
Comment (LZHgrla) [inline xtuner/configs/llava/vicuna_7b_v15_clip_vit_large_p14_336/finetune/llava_vicuna_7b_v15_qlora_clip_vit_large_p14_336_lora_e1_gpu8_finetune_refcoco.py]: Shall we use EvaluateChatHook?

File: docs/en/user_guides/dataset_prepare.md [hunk 3]
```diff
@@ -131,3 +137,46 @@ git clone https://huggingface.co/datasets/liuhaotian/LLaVA-Pretrain --depth=1
...
+| ------------ | :------------------------------------------------------------------------------------------: |
+| RefCOCO      | <a href="https://bvisionweb1.cs.unc.edu/licheng/referit/data/refcoco.zip"> annotations </a>  |
+| RefCOCO+     | <a href="https://bvisionweb1.cs.unc.edu/licheng/referit/data/refcoco+.zip"> annotations </a> |
+| RefCOCOg     | <a href="https://bvisionweb1.cs.unc.edu/licheng/referit/data/refcocog.zip"> annotations </a> |
+
+After downloading the annotations, unzip the files and place them in the `./data/refcoco_annotations` directory.
+Then, we convert the annotations to json format using the below command. This command saves the converted json files in the `./data/llava_data/RefCOCOJson/` directory.
+
+```shell
+python xtuner/tools/convert_refcoco.py
```
Comment (pppppM) [inline docs/en/user_guides/dataset_prepare.md]:
```suggestion
xtuner preprocess refcoco
```

File: docs/zh_cn/user_guides/dataset_prepare.md [hunk 4]
```diff
@@ -131,3 +137,46 @@ git clone https://huggingface.co/datasets/liuhaotian/LLaVA-Pretrain --depth=1
...
+| ------------ | :------------------------------------------------------------------------------------------: |
+| RefCOCO      | <a href="https://bvisionweb1.cs.unc.edu/licheng/referit/data/refcoco.zip"> annotations </a>  |
+| RefCOCO+     | <a href="https://bvisionweb1.cs.unc.edu/licheng/referit/data/refcoco+.zip"> annotations </a> |
+| RefCOCOg     | <a href="https://bvisionweb1.cs.unc.edu/licheng/referit/data/refcocog.zip"> annotations </a> |
+
+在下载完refcoco相关数据文件后，解压文件并将它们放在 `./data/refcoco_annotations` 目录中。
+然后，我们使用以下命令将注释转换为json格式。此命令将转换后的json文件保存在 `./data/llava_data/RefCOCOJson/` 目录中。
+
+```shell
+python xtuner/tools/convert_refcoco.py
```
Comment (pppppM) [inline docs/zh_cn/user_guides/dataset_prepare.md]:
```suggestion
xtuner preprocess refcoco
```

**Review by pppppM — APPROVED** [review]

File: xtuner/entry_point.py [hunk 5]
```diff
@@ -46,12 +47,16 @@
...
             xtuner preprocess arxiv $SRC_FILE $DST_FILE --start-date $START_DATE --categories $CATEGORIES
+        6-2. Preprocess refcoco dataset:
+            xtuner preprocess refcoco --ann-path $RefCOCO_ANN_PATH --image-path $COCO_IMAGE_PATH --save-path $SAVE_PATH
         7-1. Log processed dataset:
             xtuner log-dataset $CONFIG
         7-2. Verify the correctness of the config file for the custom dataset:
             xtuner check-custom-dataset $CONFIG
         8. MMBench evaluation:
             xtuner mmbench $LLM --llava $LLAVA --visual-encoder $VISUAL_ENCODER --prompt-template $PROMPT_TEMPLATE --data-path $MMBENCH_DATA_PATH
+        8. Refcoco evaluation:
```
Comment (LZHgrla) [inline xtuner/entry_point.py]:
```suggestion
        9. Refcoco evaluation:
```

**Review by LZHgrla — CHANGES_REQUESTED** [review]

**Review by LZHgrla — APPROVED** [review]

**Pull request merged**